### ISSUE TITLE
Frame Mismatch - Video encoding stops at the middle of the video input file

### ISSUE THREAD

**Issue by pindakaas8** [staxrip/staxrip#1242]
Hello StaxRip devs and users,

I am new to the StaxRip tool and this forum. You could call me a newbie also in Video Editing.
I found some useful Youtube tutorials showing how to use StaxRip for Deinterlacing recorded miniDV camera footage.
The recorded files were taken via WinDV and come as .avi files. I will join these files then to a single/larger .avi file via Avidemux tool.

In my first trials I took one of the small avi-clips and tried the settings in StaxRip.
I really like the results of the QTGMC Deinterlace Filter, so I configured to use it. I tried medium and slow speed.
Output video file shall also be x264 mp4 format, for audio output I use AC3, since AAC resulted in errors, because I miss some packs.

Now here is the issue that I am facing:
With QTGMC chosen as Deinterlace filter, the encoding always stops somewhere about the middle of the video file (47-50%).
I receive an error message "Frame Mismatch" for example like this: 

**WARNING: Target file has 790 frames, but should have 1592 frames!
Encoding was probably terminated at 49,6%!**

No matter what input file I choose or what speed I select for QTGMC - it always fails at the middle of the video file.

As result I get a mp4 file, which is perfect until the middle of the clip, there the last frame is then frozen until the end with perfectly fine audio (audio continues, video freezes at the last frame).

This is no coincidence... What am I doing wrong? Is there something wrong with the frame value of my input files?
I know that QTGMC doubles up the framerate from 25 to 50 fps. I even tried to edit the code of QTGMC with the expression 
", fpsdivisor=2". This sets my framerate back to 25 fps but the error while encoding stays the same - in the middle of the Job.

I also looked this up at the Issues and found that the CPU of an user was too old/ had too low performance for QTGMC.
Is this also the reason for me? But then why does it work perfectly until the exact middle of the video - every time?

Maybe someone can help looking at the logfile, I also can supply one of the clips I use. 
Here is a logfile from one avi clip and also the logfile of the large, joined avi file (where it stopped at exactly 50%):
[tape_02.06-07-17_10-09.01_staxrip.log](https://github.com/staxrip/staxrip/files/14376549/tape_02.06-07-17_10-09.01_staxrip.log)
[tape_02_alle_staxrip.log](https://github.com/staxrip/staxrip/files/14377843/tape_02_alle_staxrip.log)

And also I tried the filter Yadif2 - this worked fine and the job got completed without errors.
But tbh I want to use QTGMC still.
Pls help.

**Comment by pindakaas8** [comment]
Thanks for the support you are giving.

It's not fluctuating a lot (always between **49** (not 47, sry for the typo) and **50%**). 
It is always at (almost) the half of the frame value. I think the next encoding step would exceed the half of the frame value.

For example the large avi file: Target file has 44329 frames, but should have 88667 frames! --> half would be 44333.5
Your "Frame Mismatch" prompt calculates the termination point at 50%.
For example a smaller avi file (~1 min): Target file has 1419 frames, but should have 2854 frames! --> half would be 1427
Your "Frame Mismatch" prompt calculates the termination point at 49,7%.

I think it is no coincidence that this happens always at the middle of the file. Does insufficient amount of RAM explain this? It's hard to believe. The RAM seems to be good enough for the half of an 1h clip aswell as the half of a 1 min clip...

What can I change with SourceMatch? Shall I put another value in it? Do you have some sort of documentation for QTGMC?

Can there be something wrong with my .avi files? Inserting them as source file always sets Source=FFVideoSource -> is that right?
![grafik](https://github.com/staxrip/staxrip/assets/160884446/dff70175-f848-4abe-90fe-318cec7b7db4)
The Avidemux tool recognizes the frame type I-BFF on these avi clips. On Staxrip only "BFF" is recognized in my source.
![grafik](https://github.com/staxrip/staxrip/assets/160884446/2d38490d-6f40-4e3a-aeb0-1ff4c744de5d)

My gut tells me that the amount of frames in the Encoding Job is not processed correctly.

**Comment by Dendraspis** [comment]
I don't know what type of tapes you are encoding, but is there a chance to share one of those files, that cause the issue, with me? Via Discord you are able to send me the download link via DM if you don't want it to be *public*.

> Does insufficient amount of RAM explain this? It's hard to believe. 

Kinda yes, it depends what `QTGMC` and its dependencies do at that point.

> The RAM seems to be good enough for the half of an 1h clip aswell as the half of a 1 min clip

As I said, 32GB are sometimes not enough when it comes to using `QTGMC` on a 900p source file, so I wouldn't declare it *impossible*.

> What can I change with SourceMatch? Shall I put another value in it?

Using `SourceMatch = 0` should disable a lot of extra processing, but also make the result worse. At least for testing you should try it.

> Do you have some sort of documentation for QTGMC?

Just use Google?!  But I still had it open, so here it is: http://avisynth.nl/index.php/QTGMC

**Comment by Dendraspis** [comment]
After checkout out the source file it seems to be a tool issue, other tools crash too.
So the only solution would be to report that issue and wait for a fix or solution, whereas it might take a very long time, or to switch over to `VapourSynth` as that runs smoothly to the end.

<img width=200 src="https://github.com/staxrip/staxrip/assets/50659978/8936e702-52a2-45b0-8e21-6062d98d921e">
<img width=800 src="https://github.com/staxrip/staxrip/assets/50659978/c8cbf481-86de-4837-953d-63aa48bd8643">

**Comment by pindakaas8** [comment]
Thanks Dendraspis. I tried VapourSynth and now the Encoding works!

Just so everyone knows how to select VS Filters: Left click on Dendraspis underlined text where it says "AVS Filters", on the dropdown menu go to "Filter Setup" and select "VapourSynth" -> now Vapoursynth loads into the tool.

At first my encoded videos were very "shaky" -> thats due to the "Field" code. I needed to switch to value 1 because my input avi file is a BFF video.
Additionally on the QTGMC code I needed to replace "TFF=True" with"TFF=False" -> again, because my input is BFF.

The result is stunningly good. Even with preset "Slow" the Encoding was super quick, 25 or more fps while encoding. Nice deinterlaced video, good quality.

Again, thank you very much!
